libraries/chem-meta/src/parsing-utils/chemical-table-parser-base.ts

```diff
@@ -19,26 +9,26 @@ type CoordinateArrays = {
   z: Float32Array,
 }
 
-/** Base singleton for Molfile or Mol2 parser/handler */
-export abstract class ChemicalTableParserBase implements ChemicalTableParser {
-  protected constructor(file: string) {
-    this.init(file);
+/** Base for Molfile or Mol2 parser/handler */
+export abstract class ChemicalTableParserBase {
+  constructor(fileContent: string) {
+    this.init(fileContent);
   };
 
-  protected static instance: ChemicalTableParserBase;
-
-  protected file!: string;
+  protected fileContent!: string;
   protected _atomCount?: number;
   protected _bondCount?: number;
   protected xyzAtomCoordinates?: CoordinateArrays;
   protected _atomTypes?: string[];
   protected _pairsOfBondedAtoms?: Uint16Array[];
+  protected _bondTypes?: Uint16Array;
 
-  public init(file: string): void {
-    this.file = file.replace(/\r/g, '');
+  protected init(fileContent: string): void {
+    this.fileContent = fileContent.replace(/\r/g, '');
     this._atomCount = undefined;
     this._atomTypes = undefined;
     this._bondCount = undefined;
+    this._bondTypes = undefined;
```

Comment:
Why are we initializing it to undefined? I believe this is the default value. And in any case, if we really need it then it's better to combine declaration with initialization